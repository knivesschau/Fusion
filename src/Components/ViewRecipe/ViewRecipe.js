import React, { Component } from 'react';
import Recipe from '../Recipe/Recipe';
import fusionContext from '../../fusionContext';
import './ViewRecipe.css';

export default class ViewRecipe extends Component {
    static defaultProps = {
        match: {
            params: {}
        }
    };

    static contextType = fusionContext;

    handleDeleteRecipe = fused_id => {
        this.props.history.push(`/your-cookbook`);
    };
    
    render() {
        const {fusions=[]} = this.context;
        const {fused_id} = this.props.match.params;
        const fusionRecipe = fusions.find(fusion => fusion.fused_id === parseInt(fused_id)) || {};

        return (
            <section className="Recipe_Viewer">
                <Recipe
                    fused_id={fusionRecipe.fused_id}
                    fused_name={fusionRecipe.fused_name}
                    date_modified={fusionRecipe.date_modified}
                    base_cuisine={fusionRecipe.base_cuisine}
                    fuse_cuisine={fusionRecipe.fuse_cuisine}
                    fuse_ingredients={fusionRecipe.fuse_ingredients}
                    fuse_steps={fusionRecipe.fuse_steps}
                    onDeleteFusion={this.handleDeleteRecipe}
                />
            </section>
        );
    };
};