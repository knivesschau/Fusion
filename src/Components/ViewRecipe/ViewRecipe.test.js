import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import ViewRecipe from './ViewRecipe';

describe ('ViewRecipe Component', () => {
    it ('Renders the complete ViewRecipe Component and all nested Components', () => {
        const wrapper = shallow(<ViewRecipe/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});