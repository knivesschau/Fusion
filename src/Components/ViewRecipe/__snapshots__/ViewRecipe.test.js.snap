// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewRecipe Component Renders the complete ViewRecipe Component and all nested Components 1`] = `
<section
  className="Recipe_Viewer"
>
  <Recipe
    onDeleteFusion={[Function]}
  />
</section>
`;
